rasa/utils/tensorflow/model_data.py

```diff
@@ -19,10 +20,22 @@
 import scipy.sparse
 from sklearn.model_selection import train_test_split
 
-
 logger = logging.getLogger(__name__)
 
 
+def ragged_array_to_ndarray(ragged_array: Iterable[np.ndarray]) -> np.ndarray:
+    """Converts list (iterable) of numpy arrays to numpy array.
```

Comment:
This might be a lame question that I'm asking,
Why do we check for a ValueError to make the np.array as an dtype-object, instead why don't we just create every numpy array with dtype as object?

Reply:
This is a great question. 

We want to preserve the knowledge of type if we can when we create an array because we use the knowledge of the type in some functions to make some decisions.
For example [here](https://github.com/RasaHQ/rasa/blob/28eb117457b04399df67d1c7e49461bcd2801a84/rasa/utils/tensorflow/model_data.py#L210)

Reply:
Could you please explain in the docstring what is a `ragged_array` too?